usaspending_api/common/helpers/etl_helpers.py

```diff
@@ -59,3 +60,19 @@ def update_c_to_d_linkages(type):
     logger.info('Count of unlinked %s records after updates: %s' % (type, str(ending_unlinked_count)))
 
     logger.info('Finished all queries in %s seconds' % str(datetime.now() - total_start))
+
+
+def upsert_transactions(ids, linkage_type):
+
+    if ids:
+        update_award_ids = tuple(set(ids))  # Convert to tuple and remove duplicates.
+        with timer("updating awards to reflect their latest associated transaction info", logger.info):
+            update_awards(update_award_ids)
+        with timer("updating award category variables", logger.info):
+            update_award_categories(update_award_ids)
+
+        with timer("updating C->D linkages", logger.info):
+            update_c_to_d_linkages(linkage_type)
```

Comment:
These lines are a little different than in the other version, meaning business logic might have been changed. I can't find any such changes, but absence of evidence is not evidence of absence.

Reply:
Do you have a more accurate name to suggest? `upsert_awards`?

Reply:
Yes, I was a bit confused on the whole `upsert_transactions` accepting award ids as well.  As of right now (about 10% through my code review), I'm for renaming the function.